db/trie/merklepatriciatree/extensionnode.go

```diff
@@ -50,21 +50,18 @@ func (e *extensionNode) Delete(key keyType, offset uint8) (node, error) {
 		return nil, err
 	}
 	if newChild == nil {
-		return nil, e.mpt.deleteNode(e)
+		return nil, e.delete()
 	}
 	switch node := newChild.(type) {
 	case *extensionNode:
-		if err := e.mpt.deleteNode(e); err != nil {
```

Comment:
why this can be removed for extension node?

Reply:
node delete happens inside of the updatePath, updateChild